x-pack/legacy/plugins/siem/public/containers/case/use_get_cases.tsx

```diff
@@ -0,0 +1,159 @@
+import { AllCases, SortFieldCase, FilterOptions, QueryParams } from './types';
+import { getTypedPayload } from './utils';
+import { Direction } from '../../graphql/types';
+import { errorToToaster } from '../../components/ml/api/error_to_toaster';
+import { useStateToaster } from '../../components/toasters';
+import * as i18n from './translations';
+import { getCases } from './api';
+
+export interface UseGetCasesState {
+  data: AllCases;
+  isLoading: boolean;
+  isError: boolean;
+  queryParams: QueryParams;
+  filterOptions: FilterOptions;
+}
+
+export interface QueryArgs {
+  page?: number;
+  perPage?: number;
+  sortField?: SortFieldCase;
+  sortOrder?: Direction;
+}
+
+export interface Action {
+  type: string;
+  payload?: AllCases | QueryArgs | FilterOptions;
+}
+const dataFetchReducer = (state: UseGetCasesState, action: Action): UseGetCasesState => {
+  switch (action.type) {
+    case FETCH_INIT:
+      return {
+        ...state,
+        isLoading: true,
+        isError: false,
+      };
+    case FETCH_SUCCESS:
+      return {
+        ...state,
+        isLoading: false,
+        isError: false,
+        data: getTypedPayload<AllCases>(action.payload),
+      };
+    case FETCH_FAILURE:
+      return {
+        ...state,
+        isLoading: false,
+        isError: true,
+      };
+    case UPDATE_QUERY_PARAMS:
+      return {
+        ...state,
+        queryParams: {
+          ...state.queryParams,
+          ...action.payload,
+        },
+      };
+    case UPDATE_FILTER_OPTIONS:
+      return {
+        ...state,
+        filterOptions: getTypedPayload<FilterOptions>(action.payload),
+      };
+    default:
+      throw new Error();
+  }
+};
+
+function useDidUpdateEffect(fn: () => void, inputs: unknown[]) {
+  const didUpdateRef = useRef(false);
+
+  useEffect(() => {
+    if (didUpdateRef.current) fn();
+    else didUpdateRef.current = true;
+  }, inputs);
+}
+
+const initialData: AllCases = {
+  page: 0,
+  per_page: 0,
+  total: 0,
+  cases: [],
+};
+export const useGetCases = (): [
+  UseGetCasesState,
+  Dispatch<SetStateAction<QueryArgs>>,
+  Dispatch<SetStateAction<FilterOptions>>
+] => {
+  const [state, dispatch] = useReducer(dataFetchReducer, {
+    isLoading: false,
+    isError: false,
+    data: initialData,
+    filterOptions: {
+      search: '',
+      tags: [],
+    },
+    queryParams: {
+      page: DEFAULT_TABLE_ACTIVE_PAGE,
+      perPage: DEFAULT_TABLE_LIMIT,
+      sortField: SortFieldCase.createdAt,
+      sortOrder: Direction.desc,
+    },
+  });
+  const [queryParams, setQueryParams] = useState(state.queryParams as QueryArgs);
+  const [filterQuery, setFilters] = useState(state.filterOptions as FilterOptions);
+  const [, dispatchToaster] = useStateToaster();
+
+  useDidUpdateEffect(() => {
+    dispatch({ type: UPDATE_QUERY_PARAMS, payload: queryParams });
+  }, [queryParams]);
+
+  useDidUpdateEffect(() => {
+    dispatch({ type: UPDATE_FILTER_OPTIONS, payload: filterQuery });
```

Comment:
ok instead of doing that why we are not comparing the state.filterOptions against filterQuery and if they are not equal we are dispatching the action

Reply:
""